assembly/src/assembler/mod.rs

```diff
@@ -129,26 +95,6 @@ impl Assembler {
         assembler
     }
 
-    /// Start building an [Assembler], with a kernel given by compiling the given source module.
-    ///
-    /// # Errors
-    /// Returns an error if compiling kernel source results in an error.
-    pub fn with_kernel_from_module(module: impl Compile) -> Result<Self, Report> {
```

Comment:
Not sure if we should remove `with_kernel()` in this PR as well?

Reply:
We need a way to construct the `Assembler` with only a single kernel library, so we probably still need it. If we have metadata associated with a `Library` that tells us whether it is a kernel library or not, then we can probably make it less explicit and check that we've only added a single library of kernel type in `add_library`